HHS/Head-Start-TTADP
# Add file upload api

**Description of change**
This PR adds the a API endpoint POST /files. This is only for the api, and does not include the front end.

**How to test**
1. Update .env file with new variables: 
    1. `AWS_SECRET_ACCESS_KEY=EXAMPLEKEY`
    2. `AWS_ACCESS_KEY_ID=EXAMPLEID`
    3. `MINIO_ROOT_PASSWORD=EXAMPLKEY`
    4. `MINIO_ROOT_USER=EXAMPLEID`
    5. `S3_BUCKET=ttadp-test`
    7. If you aren't using docker, `S3_ENDPOINT=http://localhost:9000`, or if you are `S3_ENDPOINT=http://minio:9000`
2. Start app using `yarn docker:start`
2. Go to localhost:9000 to access the Minio web console (use AWS_SECRET_ACCESS_KEY for password and AWS_ACCESS_KEY_ID for username).
3. if you aren't using docker create a bucket named `ttadp-test` (It should be created automatically if you are using docker)
4. Open the web ui at `http://localhost:3000` and save a Activity Report. Note the reportId to use in your curl command.
4. Open a terminal and type `curl localhost:3000/api/files -F File=@TESTFILE.txt -F reportId=1 -F attachmentType=ATTACHMENT` where TESTFILE.txt is a file in the directory you run the curl command from
5. Output should be similar to  
```
{
"id":5
}
```
6. Check the DB to ensure the appropriate metadata was added to the `Files` table

**Issue(s)**
* part of #150 still needs front end integration
* closes #130
* closes https://github.com/HHS/Head-Start-TTADP/issues/235

**Checklist**

    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
